### PR TITLE
Merge repl BNW -> master

### PR DESCRIPTION
**Pull request by buddhisthead**
Merges the Brave New World Replication code into Master.

NOTE: 1.2 has already been re-merged into Master.

The only merge conflict (amazingly) was the rebar.config. I just needed to point it's dependency on ranch to 0.4.0 to match our BNW and Anya1.2 configurations.

Thanks for reviewing.

Cheers, Chris

### CODE REVIEW

File: ebin/riak_repl.app [hunk 1]
```diff
@@ -3,7 +3,7 @@
  riak_repl,
  [{description,  "Enterprise replication for Riak"},
   {id,           "riak_repl"},
-  {vsn,          "1.2.1"},
+  {vsn,          "1.2.3bnw"},
```
Comment (bookshelfdave) [inline ebin/riak_repl.app:6]: Is 1.2.3bnw still correct?

Comment (jaredmorrow) [inline ebin/riak_repl.app:6]: It probably doesn't matter as it will change to 1.3 soon.

**Comment by bookshelfdave** [comment]
riak_test results: https://gist.github.com/542f5443e62592235df2
+1

**Comment by seancribbs** [comment]
Does that riak_test live in the repl repo?

**Comment by bookshelfdave** [comment]
@seancribbs no, it's in the adt-repl2-test branch at the moment.

**Comment by seancribbs** [comment]
@metadave I'd like to have that either working from within a riak_ee checkout, or in riak_test proper (assuming that it doesn't leak proprietary info.) We'll need it for automated builds.

**Comment by bookshelfdave** [comment]
@seancribbs agreed, but I don't think it will be ready for 1.3

**Comment by buddhisthead** [comment]
Thank you very much @lordnull and @metadave for reviewing, fixing, testing, and merging this PR. This is a huge step for BNW in 1.3!

**Pull request merged**